lib/gooddata/lcm/actions/segments_filter.rb

```diff
@@ -29,9 +29,16 @@ def call(params)
           fail "Parameter 'segments' contains duplicate segment id(s): #{duplicated_segment_ids.join(', ')}" if duplicated_segment_ids.any?
 
           if params.segments_filter
+            data_product = params.data_product
```

Comment:
Refers to 'params' more than self (maybe move it to another class?) (FeatureEnvy)